### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by olehedza**

### CODE REVIEW

File: src/main/java/core/basesyntax/Palindrome.java [hunk 1]
```diff
@@ -21,6 +21,23 @@
      * <p>Результат: true</p>
      */
     public boolean isPalindrome(String text) {
-        return false;
+        String reverse;
+        String wholeStr;
+        String[] tokens = text.toLowerCase()
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Palindrome.java]: @olehedza this solution is to complex. Can we simplify? 
1. You use regexp to replace the symbols

File: src/main/java/core/basesyntax/Palindrome.java [hunk 2]
```diff
@@ -21,6 +21,23 @@
      * <p>Результат: true</p>
      */
     public boolean isPalindrome(String text) {
-        return false;
+        String reverse;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Palindrome.java]: @olehedza try to declare the variable as close as possible to the first usage

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Palindrome.java (hunk 2 above)
Comment (olehedza) [reply src/main/java/core/basesyntax/Palindrome.java]: Done!

File: src/main/java/core/basesyntax/Palindrome.java (hunk 1 above)
Comment (olehedza) [reply src/main/java/core/basesyntax/Palindrome.java]: Done!

File: src/main/java/core/basesyntax/Palindrome.java [hunk 3]
```diff
@@ -21,23 +21,18 @@
      * <p>Результат: true</p>
      */
     public boolean isPalindrome(String text) {
-        String reverse;
-        String wholeStr;
         String[] tokens = text.toLowerCase()
                 .replaceAll("[,.:;()!?]", "")
-                .replaceAll("(?<=[ ])[-](?=[ ])", "")
-                .replaceAll(" {2,}", " ")
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Palindrome.java]: you can do that with one replaceAll() try to find regex for it

File: src/main/java/core/basesyntax/Palindrome.java [hunk 4]
```diff
@@ -21,23 +21,18 @@
...
                 .trim()
                 .split(" ");
 
         for (String str : tokens) {
-            reverse = new StringBuilder(str).reverse().toString();
+            String reverse = new StringBuilder(str).reverse().toString();
             if (str.equals(reverse)) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Palindrome.java]: what does equals return?

Comment (olehedza) [reply src/main/java/core/basesyntax/Palindrome.java]: Fixed

File: src/main/java/core/basesyntax/Palindrome.java (hunk 3 above)
Comment (olehedza) [reply src/main/java/core/basesyntax/Palindrome.java]: Fixed

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**